src/sagemaker/base_predictor.py

```diff
@@ -0,0 +1,576 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Placeholder docstring"""
+from __future__ import print_function, absolute_import
+
+import abc
+from typing import Any, Tuple
+
+from sagemaker.deprecations import (
+    deprecated_class,
+    deprecated_deserialize,
+    deprecated_serialize,
+    removed_kwargs,
+    renamed_kwargs,
+    renamed_warning,
+)
+from sagemaker.deserializers import (  # noqa: F401 # pylint: disable=unused-import
+    BytesDeserializer,
+    CSVDeserializer,
+    JSONDeserializer,
+    NumpyDeserializer,
+    StreamDeserializer,
+    StringDeserializer,
+)
+from sagemaker.model_monitor import (
+    DataCaptureConfig,
+    DefaultModelMonitor,
+    ModelBiasMonitor,
+    ModelExplainabilityMonitor,
+    ModelMonitor,
+    ModelQualityMonitor,
+)
+from sagemaker.serializers import (
+    CSVSerializer,
+    IdentitySerializer,
+    JSONSerializer,
+    NumpySerializer,
+)
+from sagemaker.session import production_variant, Session
+from sagemaker.utils import name_from_base
+
+from sagemaker.model_monitor.model_monitoring import DEFAULT_REPOSITORY_NAME
+
+from sagemaker.lineage.context import EndpointContext
+
+
+class PredictorBase(abc.ABC):
+    """An object that encapsulates a deployed model."""
+
+    @abc.abstractmethod
+    def predict(self, *args, **kwargs) -> Any:
+        """Perform inference on the provided data and return a prediction."""
+
+    @abc.abstractmethod
+    def delete_predictor(self, *args, **kwargs) -> None:
+        """Destroy resources associated with this predictor."""
+
+    @property
+    @abc.abstractmethod
+    def content_type(self) -> str:
+        """The MIME type of the data sent to the inference server."""
+
+    @property
+    @abc.abstractmethod
+    def accept(self) -> Tuple[str]:
+        """The content type(s) that are expected from the inference server."""
+
+
+class Predictor(PredictorBase):
```

Comment:
This movement of code changes accross modules are backwards incompatible, and might code like - https://github.com/aws/amazon-sagemaker-examples/blob/692c23949ddddc8abdeca7e58ddde85535f35ffd/advanced_functionality/autogluon-sagemaker-pipeline/pipelines/ag_model.py#L7
I can't allow this.

Reply:
Hey, on the surface it looks backwards incompatible. But if you look at the top of predictors.py, we import everything from base_predictors.py. You can see that the unit/integ tests pass which means this is backwards compatible.